nilearn/_utils/cache_mixin.py

```diff
@@ -27,7 +28,7 @@ def _check_memory(memory, verbose=0):
 
     Parameters
     ----------
-    memory : None or instance of joblib.Memory or str
+    memory : None or instance of joblib.Memory or str or path-like object
```

Comment:
maybe we could add "path-like" to the glossary? alternatively, we can just call them `pathlib.Path` as this will be the vast majority of "path-like" object actually used by nilearn users